### PR TITLE
[TS] LPS-149028

### PR DESCRIPTION
**Pull request by joshuacords**
https://issues.liferay.com/browse/LPS-149028

Search Team PR here: https://github.com/liferay-search/liferay-portal/pull/331

Expando number fields need to be indexed as numbers in order for Segmentation to properly search them with criteria.

**Author:** @joshuacords

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portlet/expando/util/ExpandoBridgeIndexerImpl.java [hunk 1]
```diff
@@ -126,25 +126,41 @@ else if (type == ExpandoColumnConstants.DATE) {
...
 			}
 		}
 		else if (type == ExpandoColumnConstants.FLOAT) {
-			document.addKeyword(fieldName, expandoValue.getFloat());
+			document.addNumberSortable(fieldName, expandoValue.getFloat());
 		}
 		else if (type == ExpandoColumnConstants.FLOAT_ARRAY) {
 			if (!defaultValue) {
-				document.addKeyword(fieldName, expandoValue.getFloatArray());
+				float[] floatValues = expandoValue.getFloatArray();
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portlet/expando/util/ExpandoBridgeIndexerImpl.java]: @joshuacords Please use ArrayUtil

File: portal-impl/src/com/liferay/portlet/expando/util/ExpandoBridgeIndexerImpl.java [hunk 2]
```diff
@@ -157,50 +173,68 @@ else if (type == ExpandoColumnConstants.GEOLOCATION) {
...
 		}
 		else if (type == ExpandoColumnConstants.SHORT) {
-			document.addKeyword(fieldName, expandoValue.getShort());
+			document.addNumberSortable(fieldName,
+				new Long(expandoValue.getShort()));
 		}
 		else if (type == ExpandoColumnConstants.SHORT_ARRAY) {
 			if (!defaultValue) {
-				document.addKeyword(fieldName, expandoValue.getShortArray());
+				short[] shortValues = expandoValue.getShortArray();
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portlet/expando/util/ExpandoBridgeIndexerImpl.java]: @joshuacords Why a short array translate to a Long array, I know there is not API for short array, why not using Integer array which is closer to short

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/UserExpandoColumnModelListener.java [hunk 3]
```diff
@@ -201,7 +201,8 @@ else if ((expandoColumn.getType() == ExpandoColumnConstants.DOUBLE) ||
 					 ExpandoColumnConstants.FLOAT_ARRAY)) {
 
 			entityField = new DoubleEntityField(
-				encodedName, locale -> encodedIndexedFieldName);
+				encodedName,
```
Comment (tinatian) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/UserExpandoColumnModelListener.java:204]: @joshuacords Please do not hard code logic here, please move the logic to ExpandoBridgeIndexerImpl#encodeFieldName

**Comment by joshuacords** [comment]
Thanks Tina

**Pull request closed without merging**